.teamcity/_self/PluginBaseBuild.kt

```diff
@@ -59,6 +59,24 @@ open class PluginBaseBuild : Template({
 		bashNodeScript {
 			name = "Prepare environment"
 			scriptContent = """
+				# Merge the trunk branch first. This way, our builds and tests
+				# include the latest merged version of the plugin being built.
+				# Otherwise, we can get into a situation where the current plugin
+				# build appears "different", but that's just because it's older.
+				if [[ "%teamcity.build.branch.is_default%" != "true" ]] ; then
+					# git operations will fail if no user is set.
+					git config --local user.email "tcbuildagent@example.com"
+					git config --local user.name "TeamCity Build Agent"
+					# Note that `trunk` is already up-to-date from the `teamcity.git.fetchAllHeads`
+					# parameter in the project settings.
+					if ! git merge trunk ; then
+						echo "##teamcity[buildProblem description='There is a merge conflict with trunk. Rebase on trunk to resolve this problem.' identity='merge_conflict']]"
```

Comment:
TeamCity has many drawbacks, but I love how [Service Messages](https://www.jetbrains.com/help/teamcity/service-messages.html) make this kind of notifications really easy

Reply:
Yeah, it's a great feature!